app/components/UI/Tokens/TokenList/TokenListItem/index.tsx

```
@@ -159,17 +165,24 @@ export const TokenListItem = React.memo(
       ],
     );
 
-    const tokenPercentageChange = asset.address
-      ? multiChainMarketData?.[chainId as Hex]?.[asset.address as Hex]
-          ?.pricePercentChange1d
-      : undefined;
+    let pricePercentChange1d;
 
-    const pricePercentChange1d = asset.isNative
-      ? multiChainMarketData?.[chainId as Hex]?.[
-          getNativeTokenAddress(chainId as Hex) as Hex
-        ]?.pricePercentChange1d
-      : tokenPercentageChange;
+    if (isEvmNetworkSelected) {
+      const tokenPercentageChange = asset.address
+        ? multiChainMarketData?.[chainId as Hex]?.[asset.address as Hex]
+            ?.pricePercentChange1d
+        : undefined;
 
+      pricePercentChange1d = asset.isNative
+        ? multiChainMarketData?.[chainId as Hex]?.[
+            getNativeTokenAddress(chainId as Hex) as Hex
+          ]?.pricePercentChange1d
+        : tokenPercentageChange;
+    } else {
+      pricePercentChange1d =
+        allMultichainAssetsRates[asset?.address as CaipAssetType]?.marketData
+          ?.pricePercentChange?.P1D;
```

Since the above variable is codefenced, should this also be code-fenced?

---

I would assume we'd at least need a `!isEvmNetwork` check

---

hmm interesting!
I all for ditching codefences (they are so painful to debug lol)

---

I am surprised this is working as is. Given that the `allMultichainAssetsRates` is behind a code fence, this variable would not exist on a main build and should throw an error?

---

Actually i dnt think code fence is needed here; we should instead only check if isEvmNetworkSelected then display the EVM percent change; else display the non EVM percent change.

---

Refactored here [08e6bd6](https://github.com/MetaMask/metamask-mobile/pull/14892/commits/08e6bd69e92f21100b056b06a50dbcef6208b227)